packages/types/src/products.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * - `null` - explicitly Channels
+ * - `string` - uuid - any other product
+ */
+export type ProductIdentifier = null | string;
+
+/** @see {@link ProductIdentifier} */
+export type ProductScope = ProductIdentifier | ProductIdentifier[];
+
+export const inScope = (scope: ProductScope, productId: ProductIdentifier, pluginId?: string) => {
```

Comment:
Does this need to live here, or can it be kept in the web app? I'm not sure where we would use this that isn't just in the web app. 0/5 though

Reply:
While I was drafting these, the thought was perhaps usage in plugins/products, but I don't think that's too likely. Slightly in favor of boolean-utils co-located with the types, but also 0/5 so I'll just go ahead and move them back.

Reply:
Sounds good to me. We really need to get around to figuring out where those sort of shared utilities go one of these days...

Reply:
^ 💯